### PR TITLE
Shut up some Coverity warnings

### PR DESCRIPTION
**Pull request by dirkhh**
### Describe the pull request:

- [x] Code cleanup

### Pull request long description:

mostly in the 'noise' category.

### CODE REVIEW

File: core/trip.c [hunk 1]
```diff
@@ -74,10 +74,12 @@ bool is_trip_before_after(const struct dive *dive, bool before)
 {
 	int idx = get_idx_by_uniq_id(dive->id);
 	if (before) {
-		if (idx > 0 && get_dive(idx - 1)->divetrip)
+		struct dive *d = get_dive(idx - 1);
```
Comment (bstoeger) [inline core/trip.c]: You could write this as `const struct dive *`, to make clear that die is not modified.

File: core/trip.c [hunk 2]
```diff
@@ -74,10 +74,12 @@ bool is_trip_before_after(const struct dive *dive, bool before)
...
+		if (d && d->divetrip)
 			return true;
 	} else {
-		if (idx < dive_table.nr - 1 && get_dive(idx + 1)->divetrip)
+		struct dive *d = get_dive(idx + 1);
+		if (d && d->divetrip)
```
Comment (bstoeger) [inline core/trip.c:82]: ditto

File: qt-models/divetripmodel.cpp [hunk 3]
```diff
@@ -717,7 +717,7 @@ void DiveTripModelTree::populate()
 	for (int i = 0; i < dive_table.nr; ++i) {
 		dive *d = get_dive(i);
 		update_cylinder_related_info(d);
```
Comment (bstoeger) [inline qt-models/divetripmodel.cpp:721]: If `d` were NULL, wouldn't `update_cylinder_related_info()` crash? I.e. shouldn't the null-pointer check be moved in front of this calls? As it is, it makes no sense...

Comment (dirkhh) [reply qt-models/divetripmodel.cpp:721]: `update_cylinder_related_info()` does its own NULL check...
This way it was less of a code change. But for clarity, year, you're right.

File: core/trip.c (hunk 1 above)
Comment (dirkhh) [reply core/trip.c]: good! will do

**Review by bstoeger — APPROVED** [review]
lgtm

**Comment by dirkhh** [comment]
merging locally

**Pull request closed without merging**